File: leetcode/maximum-subarray.rb
# @param {Integer[]} nums
# @return {Integer}
def max_sub_array(nums)
    curr = 0
    max = nums[0]
    nums.each do |ele|
        
        curr += ele
        max = curr if curr > max
        curr = 0 if curr < 0
    
    end
    return max
    
end
# [-2,1,-3,4,-1,2,1,-5,4]

File: leetcode/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {

  let curr = 0;
  let max = nums[0];

  for (let i = 0; i < nums.length; i++) {

    curr += nums[i]
    if (curr > max) max = curr;
    if (curr < 0) curr = 0;

  }
  return max;

};